File: Localization.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace BitDoFixer
{
    public class Localization : INotifyPropertyChanged
    {
        public static Localization Instance { get; } = new Localization();

        private bool _isEnglish = true;
        public bool IsEnglish
        {
            get => _isEnglish;
            set
            {
                if (_isEnglish != value)
                {
                    _isEnglish = value;
                    OnPropertyChanged(null); // Notify all properties
                }
            }
        }

        // XAML Bound Properties
        public string AppTitle => IsEnglish ? "8BitDo Ultimate 2C Fixer" : "8BitDo Ultimate 2C Düzeltici";
        public string ConnectionStatusTitle => IsEnglish ? "CONTROLLER STATUS" : "KONTROLCÜ DURUMU";
        public string BatteryLevelTitle => IsEnglish ? "BATTERY" : "PİL";
        public string InputModeTitle => IsEnglish ? "TRANSLATION" : "DÖNÜŞÜM";
        public string InputModeDesc => "DirectInput ➔ XInput (Xbox 360)";
        public string StartServiceBtn => IsEnglish ? "Start Service" : "Servisi Başlat";
        public string StopServiceBtn => IsEnglish ? "Stop Service" : "Servisi Durdur";
        public string MinimizeToTrayBtn => IsEnglish ? "Minimize to Tray" : "Tepsiye Küçült";
        public string ToggleLogsText => IsEnglish ? "Logs & Diagnostics" : "Günlükler ve Tanılama";
        public string ClearLogsBtn => IsEnglish ? "Clear" : "Temizle";
        public string FooterText => "v0.1.0 • github.com/bezelye404";
        public string TrayOpen => IsEnglish ? "Open" : "Aç";
        public string TrayExit => IsEnglish ? "Exit" : "Çıkış";
        
        // Dynamic Texts (Used in Code-Behind)
        public string SearchingDInput => IsEnglish ? "Searching for controller..." : "Kontrolcü aranıyor...";
        public string NoDevice => IsEnglish ? "No Device Detected" : "Cihaz Algılanmadı";
        public string Scanning => IsEnglish ? "Scanning..." : "Taranıyor...";
        public string Connected => IsEnglish ? "Connected & Active" : "Bağlı ve Aktif";
        public string Stopped => IsEnglish ? "Service Stopped" : "Servis Durduruldu";
        public string Idle => IsEnglish ? "Idle" : "Boşta";
        public string WaitingReconnect => IsEnglish ? "Waiting for controller..." : "Kolun açılması bekleniyor...";

        // Log messages
        public string LogAppInit => IsEnglish ? "Application ready." : "Uygulama hazır.";
        public string LogServicesStarting => IsEnglish ? "Starting services..." : "Servisler başlatılıyor...";
        public string LogServicesStopping => IsEnglish ? "Stopping services..." : "Servisler durduruluyor...";
        
        public string LogMapperStart => IsEnglish ? "Bluetooth (DInput) -> Virtual Xbox 360 Remapper started." : "Bluetooth (DInput) -> Sanal Xbox 360 Remapper başlatıldı.";
        public string LogMapperNotFound => IsEnglish ? "Controller not found. Entering auto-reconnect scan..." : "Kontrolcü bulunamadı. Otomatik arama moduna geçiliyor...";
        public string MapperNotFoundStatus => IsEnglish ? "Not Found" : "Bulunamadı";
        public string LogMapperSource(string name) => IsEnglish ? $"Connected to device: {name}" : $"Cihaza bağlanıldı: {name}";
        public string MapperConnectedStatus => IsEnglish ? "Connected" : "Bağlandı";
        public string LogMapperReady => IsEnglish ? "Virtual Xbox 360 controller active and ready." : "Sanal Xbox 360 kontrolcüsü aktif ve hazır.";
        public string LogMapperError(string ex) => IsEnglish ? $"Connection dropped: {ex}. Reconnecting..." : $"Bağlantı koptu: {ex}. Yeniden bağlanılıyor...";
        public string MapperDisconnectedStatus => IsEnglish ? "Disconnected" : "Bağlantı Koptu";

        public string LogBatteryStart(int init, int interval) => IsEnglish ? $"Battery monitor active (refresh: {interval}s)." : $"Pil monitörü aktif (yenileme: {interval}sn).";
        public string LogBatteryFatal(string ex) => IsEnglish ? $"Battery monitor error: {ex}" : $"Pil monitörü hatası: {ex}";
        public string LogBatteryLevel(string name, int level) => IsEnglish ? $"{name}: {level}%" : $"{name}: %{level}";
        public string LogBatteryScanError(string ex) => IsEnglish ? $"Battery scan warning: {ex}" : $"Pil tarama uyarısı: {ex}";

        public event PropertyChangedEventHandler? PropertyChanged;
        protected void OnPropertyChanged([CallerMemberName] string? name = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
        }
    }
}

File: src/Localization.cpp
#include "Localization.h"

namespace BitDoFixer {

std::wstring Localization::Get(StringId id) const {
    if (m_isEnglish) {
        switch (id) {
            case StringId::AppTitle: return L"8BitDo Ultimate 2C Fixer";
            case StringId::StatusTitle: return L"CONTROLLER STATUS";
            case StringId::BatteryTitle: return L"BATTERY";
            case StringId::ModeDesc: return L"DirectInput -> XInput (Xbox 360)";
            case StringId::StartBtn: return L"Start Service";
            case StringId::StopBtn: return L"Stop Service";
            case StringId::TrayBtn: return L"Minimize to Tray";
            case StringId::LogsTitle: return L"Terminal & Diagnostics";
            case StringId::ClearBtn: return L"Clear";
            case StringId::Footer: return L"";
            case StringId::MinimizeOnClose: return L"Minimize to tray on close";
            case StringId::TrayOpen: return L"Open";
            case StringId::TrayExit: return L"Exit";
            case StringId::StatusSearching: return L"Searching for controller...";
            case StringId::StatusConnected: return L"Connected & Active";
            case StringId::StatusDisconnected: return L"Disconnected";
            case StringId::StatusStopped: return L"Service Stopped";
            case StringId::NoDevice: return L"No Device Detected";
            case StringId::Idle: return L"Idle";
            case StringId::WaitingReconnect: return L"Waiting for controller...";
            case StringId::LogAppReady: return L"Application ready (C++ Native).";
            case StringId::LogServicesStarting: return L"Starting remapper and battery services...";
            case StringId::LogServicesStopping: return L"Stopping services...";
            case StringId::LogMapperStart: return L"DirectInput -> Virtual Xbox 360 remapper started.";
            case StringId::LogMapperReady: return L"Virtual Xbox 360 controller connected and ready.";
            case StringId::LogMapperDisconnected: return L"Controller disconnected. Entering auto-reconnect scan...";
            case StringId::LogBatteryActive: return L"Battery monitor active.";
            case StringId::LogBatteryError: return L"Battery monitor scan warning.";
            default: return L"";
        }
    } else {
        switch (id) {
            case StringId::AppTitle: return L"8BitDo Ultimate 2C D\x00FCzeltici";
            case StringId::StatusTitle: return L"KONTROLC\x00DC DURUMU";
            case StringId::BatteryTitle: return L"P\x0130L";
            case StringId::ModeDesc: return L"DirectInput -> XInput (Xbox 360)";
            case StringId::StartBtn: return L"Servisi Ba\x015Flat";
            case StringId::StopBtn: return L"Servisi Durdur";
            case StringId::TrayBtn: return L"Tepsiye K\x00FC\x00E7\x00FClt";
            case StringId::LogsTitle: return L"Terminal ve Tan\x0131lama";
            case StringId::ClearBtn: return L"Temizle";
            case StringId::Footer: return L"";
            case StringId::MinimizeOnClose: return L"Kapat\x0131ld\x0131\x011F\x0131nda tepsiye k\x00FC\x00E7\x00FClt";
            case StringId::TrayOpen: return L"A\x00E7";
            case StringId::TrayExit: return L"\x00C7\x0131k\x0131\x015F";
            case StringId::StatusSearching: return L"Kontrolc\x00FC aran\x0131yor...";
            case StringId::StatusConnected: return L"Ba\x011Fl\x0131 ve Aktif";
            case StringId::StatusDisconnected: return L"Ba\x011Flant\x0131 Koptu";
            case StringId::StatusStopped: return L"Servis Durduruldu";
            case StringId::NoDevice: return L"Cihaz Alg\x0131lanmad\x0131";
            case StringId::Idle: return L"Bo\x015Fta";
            case StringId::WaitingReconnect: return L"Kolun a\x00E7\x0131lmas\x0131 bekleniyor...";
            case StringId::LogAppReady: return L"Uygulama haz\x0131r (Yerel C++).";
            case StringId::LogServicesStarting: return L"Servisler ba\x015Flat\x0131l\x0131yor...";
            case StringId::LogServicesStopping: return L"Servisler durduruluyor...";
            case StringId::LogMapperStart: return L"DirectInput -> Sanal Xbox 360 remapper ba\x015Flat\x0131ld\x0131.";
            case StringId::LogMapperReady: return L"Sanal Xbox 360 kontrolc\x00FCs\x00FC haz\x0131r.";
            case StringId::LogMapperDisconnected: return L"Ba\x011Flant\x0131 koptu. Otomatik yeniden ba\x011Flanma moduna ge\x00E7ildi...";
            case StringId::LogBatteryActive: return L"Pil monit\x00F6r\x00FC aktif.";
            case StringId::LogBatteryError: return L"Pil tarama uyar\x0131s\x0131.";
            default: return L"";
        }
    }
}

} // namespace BitDoFixer
